### PR TITLE
caib: support source_glob

### PR DESCRIPTION
**Pull request by bennyz**
Support uploads with source_glob

## Type of Change
- [x] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation update
- [ ] CI/CD improvement
- [ ] Refactoring

## Testing
- [x] Unit tests pass (`make test`)
- [x] Linter passes (`make lint`)
- [x] Manifests are up to date (`make manifests generate`)
- [x] Tested on OpenShift cluster (if applicable)

## Summary by CodeRabbit

* **New Features**
  * Support for glob patterns in manifests so multiple files can be referenced with wildcards.

* **Improvements**
  * Relative paths are resolved against the manifest’s directory for more reliable handling.
  * Build requests now detect and mark presence of local files earlier so uploads are determined correctly.
  * Manifest preprocessing no longer rewrites glob entries to absolute paths.

* **Tests**
  * Added tests covering glob expansion, recursive ** patterns, matching rules, and edge cases.

### CODE REVIEW

File: internal/buildapi/server.go [hunk 1]
```diff
@@ -2373,6 +2360,63 @@ func getBuildTemplate(c *gin.Context, name string) {
...
+	} `yaml:"qm"`
+}
+
+// manifestNeedsUpload parses the manifest YAML and returns true if any
+// add_files entry references local files (source_path or source_glob).
+func manifestNeedsUpload(manifest string) bool {
+	var m manifestSchema
+	if err := yaml.Unmarshal([]byte(manifest), &m); err != nil {
+		return false
+	}
```
Comment (bkhizgiy) [inline internal/buildapi/server.go:2393]: That can cover issues with the manifest parsing and skip upload even if needed, should we at least log this?

Comment (bennyz) [reply internal/buildapi/server.go:2393]: ack

**Review by bkhizgiy — APPROVED** [review]
